src/main/java/picard/vcf/LiftoverVcf.java

```diff
@@ -210,16 +219,24 @@ protected int doWork() {
             ++total;
             final Interval source = new Interval(ctx.getContig(), ctx.getStart(), ctx.getEnd(), false, ctx.getContig() + ":" + ctx.getStart() + "-" + ctx.getEnd());
             final Interval target = liftOver.liftOver(source, LIFTOVER_MIN_MATCH);
+
+            if (target == null) {
+                rejectVariant(ctx, FILTER_NO_TARGET);
+                continue;
+            }
+            if (ctx.getReference().length() != target.length()){
+                rejectVariant(ctx, FILTER_INDEL_STRADDLES_TWO_LINKS);
+                continue;
+            }
+
             final ReferenceSequence refSeq;
 
             // if the target is null OR (the target is reverse complemented AND the variant is a non-biallelic indel or mixed), then we cannot lift it over
-            if (target == null || (target.isNegativeStrand() && (ctx.isMixed() || (ctx.isIndel() && !ctx.isBiallelic())))) {
-                final String reason = (target == null) ? FILTER_NO_TARGET : FILTER_CANNOT_LIFTOVER_INDEL;
-                rejects.add(new VariantContextBuilder(ctx).filter(reason).make());
-                failedLiftover++;
+            if (target.isNegativeStrand() && (ctx.isMixed() || ctx.isIndel() && !ctx.isBiallelic())) {
+                rejectVariant(ctx, FILTER_CANNOT_LIFTOVER_INDEL);
+
             } else if (!refSeqs.containsKey(target.getContig())) {
-                rejects.add(new VariantContextBuilder(ctx).filter(FILTER_NO_TARGET).make());
```

Comment:
Just as a side point, maybe it's worth emitting a warning here?  Shouldn't this only happen when the chain file is bad?